api/src/main/java/org/pmiops/workbench/db/dao/UserService.java

```diff
@@ -101,8 +101,9 @@ private User updateWithRetries(Function<User, User> userModifier, User user) {
 
   private void updateDataAccessLevel(User user) {
     boolean shouldBeRegistered = Optional.ofNullable(user.getIdVerificationIsValid()).orElse(false)
+        && (!(configProvider.get().access.enableEraCommons) || user.getEraCommonsCompletionTime() != null)
```

Comment:
This is starting to look really clean – thanks for adding the enableComplianceTraining env flag while you're here, it really helps to set the pattern. 👍